code/modules/xenoarchaeology/xenoartifact.dm

```diff
@@ -0,0 +1,470 @@
+	traits += new new_trait
+	new_trait = new new_trait //type converting doesn't work too well here but this should be fine.
+	blacklist_traits += new_trait.blacklist_traits //Cant use initial() to access lists without bork'ing it
+	return new_trait
+	
+///Gets a singular entity, there's a specific traits that handles multiple.
+/obj/item/xenoartifact/proc/get_target_in_proximity(range)
+	for(var/mob/living/M in oview(range, get_turf(src)))
+		. = process_target(M)
+	if(isliving(loc) && !.)
+		. = loc
+	return
+
+///Returns the desired trait and it's values if it's in the artifact's list
+/obj/item/xenoartifact/proc/get_trait(typepath)
+	return (locate(typepath) in traits)
+
+///Used for hand-holding secret technique. Pulling entities swaps them for you in the target list.
+/obj/item/xenoartifact/proc/process_target(atom/target)
+	if(ishuman(target)) //early return if deflect chance
+		var/mob/living/carbon/human/H = target
+		if(H.wear_suit && H.head && isclothing(H.wear_suit) && isclothing(H.head))
+			var/obj/item/clothing/CS = H.wear_suit
+			var/obj/item/clothing/CH = H.head
+			if(((CS.clothing_flags & BLOCK_ARTIFACT)||(CH.clothing_flags & BLOCK_ARTIFACT)) && prob(XENOA_DEFLECT_CHANCE))
+				to_chat(target, "<span class='warning'>The [name] was unable to target you!.</span>")
+				playsound(get_turf(target), 'sound/weapons/deflect.ogg', 35, TRUE) 
+				return
+
+	if(isliving(target)) //handle pulling
+		var/mob/living/M = target
+		. = M?.pulling ? M.pulling : M
+	else
+		. = target
+	RegisterSignal(., COMSIG_PARENT_QDELETING, .proc/on_target_del, .)
+	return
+
+///Hard del handle
+/obj/item/xenoartifact/proc/on_target_del(atom/target)
+	UnregisterSignal(target, COMSIG_PARENT_QDELETING)
+	true_target -= target
+	target = null
+
+///Helps show how the artifact is working. Hint stuff. Draws a beam between artifact and target
+/obj/item/xenoartifact/proc/create_beam(atom/target)
+	if((locate(src) in target?.contents) || !get_turf(target))
+		return
+	var/datum/beam/xenoa_beam/B = new(src.loc, target, time=1.5 SECONDS, beam_icon='icons/obj/xenoarchaeology/xenoartifact.dmi', beam_icon_state="xenoa_beam", btype=/obj/effect/ebeam/xenoa_ebeam)
+	B.set_color(material)
+	INVOKE_ASYNC(B, /datum/beam/xenoa_beam.proc/Start)
+
+///Default template used to interface with activator signals.
+/obj/item/xenoartifact/proc/default_activate(chr, mob/user, atom/target)
+	if(!COOLDOWN_FINISHED(src, xenoa_cooldown))
+		return FALSE
+	charge = chr
+	true_target |= process_target(target)
+	check_charge(user)
+	return TRUE
+
+///Add extra icon overlays. Ghetto GAGS.
+/obj/item/xenoartifact/proc/generate_icon(var/icn, var/icnst = "", colour)
+	icon_overlay = mutable_appearance(icn, icnst)
+	icon_overlay.layer = FLOAT_LAYER //Not doing this fucks the object icons when you're holding it
+	icon_overlay.appearance_flags = RESET_ALPHA// Not doing this fucks the alpha?
+	icon_overlay.alpha = alpha
+	if(colour)
+		icon_overlay.color = colour
+	add_overlay(icon_overlay)
```

Comment:
Something in here is making things a bit funky with the pickup animation, the icon seems to persist for longer than it should.
I think the RESET_ALPHA flag is making it so the pickup animation doesn't fade away, but just disappears after being fully transparent for some time. Might be worth looking into.
Another thing to note is that you might be overriding some default appearance flags, if there are any.